ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.BusinessLogic/Validators/CreateProjectDTOValidator.cs

```diff
@@ -1,18 +1,32 @@
 ﻿namespace ExadelTimeTrackingSystem.Data.Validators
 {
     using ExadelTimeTrackingSystem.BusinessLogic.DTOs;
+    using ExadelTimeTrackingSystem.BusinessLogic.Services.Abstract;
     using FluentValidation;
 
     public class CreateProjectDTOValidator : AbstractValidator<CreateProjectDTO>
     {
-        public CreateProjectDTOValidator()
+        private readonly IUserService _userservice;
+
+        public CreateProjectDTOValidator(IUserService userservice)
+        {
+            _userservice = userservice;
+            this.ConfigureRules();
+        }
+
+        private void ConfigureRules()
         {
             RuleFor(p => p.Name)
                 .NotEmpty()
                 .MaximumLength(50);
 
             RuleFor(p => p.ApproverId)
-                .NotEmpty();
+                .NotEmpty()
+                .MustAsync(async (id, cancellation) =>
```

Comment:
you can place into MustAsync just `_userservice.ExistsAsync` - name of method

Reply:
when you use some custom rule with Must or MustAsync, you should write some custom error code in case of failed validation. Use WithErrorMessage method for it. And Create Constants static class, and Validation subclass in it, and then create some const strings for messages